proof_generation/helper/gateway.js

```diff
@@ -0,0 +1,462 @@
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const EventDecoder = require('../../test/test_lib/event_decoder');
+
+/**
+ * Object contains input parameter needed for confirmRedeemIntent.
+ *
+ * @typedef {object} params.
+ * @property {string} params.redeemer Redeemer address.
+ * @property {string} params.redeemerNonce Redeemer nonce.
+ * @property {string} params.beneficiary Address where the redeemed tokens
+ *                                    will be transferred.
+ * @property {string} params.amount Redeem amount.
+ * @property {string} params.gasPrice Gas price that redeemer is ready to pay to
+ *                             get the redeem and unstake process done.
+ * @property {string} params.gasLimit Gas limit that redeemer is ready to pay.
+ * @property {string} params.blockHeight Block number for which the proof is valid.
+ * @property {string} params.hashLock Hash lock.
+ * @property {string} params.rlpParentNodes RLP encoded proof data.
+ * @property {string} params.storageRoot Storage root for proof.
+ * @property {string} params.facilitator Facilitator address for progress mint.
+ */
+
+/**
+ * Gateway class provides helper methods for stake and mint and redeem and unstake.
+ */
+class Gateway {
+  /**
+   * @param {Object} registeredContracts All the deployed contracts
+   */
+  constructor(registeredContracts) {
+    this.token = registeredContracts.mockToken;
+    this.baseToken = registeredContracts.baseToken;
+    this.gateway = registeredContracts.gateway;
+    this.deployer = registeredContracts.owner;
+    this.address = this.gateway.address;
+  }
+
+  /**
+   * Initiates the stake process and returns the event and return value of
+   * stake function.
+   *
+   * @param {Object} params.
+   * @param {BN} params.amount Stake amount that will be transferred from
```

Comment:
These properties shouldn't be redefined, they are already defined in typedef.

Reply:
Agree! Removed them.